File: Leet 3379/solution.cs
public class Solution 
{
    private int Wrapi(int value, int lo, int hi)
    {
        return lo + ((value - lo) % (hi - lo) + (hi - lo)) % (hi - lo);
    }

    public int[] ConstructTransformedArray(int[] nums) 
    {
        int[] result = new int[nums.Length];
        for(int i = 0; i < nums.Length; i++)
        {
            result[i] = nums[Wrapi(i + nums[i], 0, nums.Length)];
        }
        return result;
    }
}

class Program
{
    static void PrintArray(int[] nums)
    { 
        Console.WriteLine($"[{string.Join(", ", nums)}]");
    }

    static void Main()
    {
        Console.WriteLine("3379. Transformed Array");
        Solution solution = new();
        int[] case1 = solution.ConstructTransformedArray([3,-2,1,1]);
        PrintArray(case1);
    }
}

File: Leet 3379/solution.py
class Solution:
    def constructTransformedArray(self, nums: list[int]) -> list[int]:
        def wrapi(value: int, lo: int, hi: int) -> int:
            return lo + ((value - lo) % (hi - lo) + (hi - lo)) % (hi - lo);

        n: int = len(nums)
        result: list[int] = [0] * n
        for i, num in enumerate(nums):
            result[i] = nums[wrapi(i + num, 0, n)]
        return result
    

def main() -> None:
    print('3379. Transformed Array')
    solution = Solution()
    test_cases = [
        [3, -2, 1, 1],
        [-1, 4, -1]
    ]
    for nums in test_cases:
        print(solution.constructTransformedArray(nums))


if __name__ == '__main__':
    main()